src/main/java/io/vertx/core/json/JsonArray.java

```diff
@@ -709,11 +703,19 @@ public int readFromBuffer(int pos, Buffer buffer) {
   }
 
   private void fromJson(String json) {
-    list = Json.CODEC.fromString(json, List.class);
+    JsonArray decoded = (JsonArray) Json.CODEC.fromString(json);
```

Comment:
why can't we specify here the `List` type anymore ? it seems fine to ask for a `List`

Reply:
If the object mapper is configured with the Jackson Scala module, and you run:

```java
var json = """
[1,2,{"foo":"bar}]
""";
JsonArray arr = Json.CODEC.fromString(json, List.class);
JsonObject obj = arr.getJsonObject(2); // boom
```

You will get a `ClassCastException` because the object nested in the array will be deserialized as a Scala Map, not a Java Map.

Reply:
so you are saying that the following code in `DatabindCodec`:

```java
      value = DatabindCodec.mapper.readValue(parser, type);
```

called with type = `Map.class` will throw this CCE ?

Reply:
Assuming the object mapper has been configured with the Jackson Scala module, if you have a `parser` for a valid json object string or buffer, and invoke:

```
      value = DatabindCodec.mapper.readValue(parser, type);
```

with the `type` of `Map`, then the `value` will be an instance of Java `Map`, because, Jackson knows the target type for the top-level object. But if you have nested objects or arrays, they will deserialized to Scala Map or Sequence, because Jackson doesn't know which type they should be deserialized to.

Reply:
have you checked we cannot override the default collection type to map when vertx takes control of the mapper ?

Reply:
Yes, and to my knowledge it's not possible.